### ISSUE TITLE
Error: no mixin named box-shadow

### ISSUE THREAD

**Issue by DaddyWarbucks** [thomaspark/bootswatch#597]
I am having trouble getting the Paper theme to compile properly because there is no mixin named box-shadow, so any inclusion of it fails

```
.navbar {
  border: none;
  @include box-shadow(0 1px 2px rgba(0,0,0,.3));
...
```

This is also failing for `Error: no mixin named placeholder`
Any help is appreciated, thanks!

**Comment by thomaspark** [comment]
Thanks, this was also mentioned in #480. Since the box-shadow vendor prefixes are no longer needed, I'll be removing use of the mixin from all themes.

**Comment by DaddyWarbucks** [comment]
If I am not mistaken, I believe that would just entail replacing `@include box-shadow(inset 0 -1px 0 #ddd)` with `box-shadow: inset 0 -1px 0 #ddd`. Correct? I wouldn't mind handling that and submitting a PR.

Nice work on all of these BTW! I built an app called stackedsports.com with this. Not the marketing site at that address, but the actual app that site is talking about.  And a few others too, so I'd be happy to make a contribution!

**Comment by thomaspark** [comment]
@DaddyWarbucks, cool that you've found such use for the themes! Would be great if you handled this issue.

The best way to go about it is to remove the `.box-shadow()` mixins in each theme's `bootswatch.less` file, then build the CSS and SASS versions by running `grunt` and `grunt convert_less` respectively in command line.

Let me know if you have any questions about it.

**Comment by DaddyWarbucks** [comment]
Oh I see. That is cool you manage it in one language and convert it to the other. I will do that and submit a PR next week probably.

**Comment by DaddyWarbucks** [comment]
I am having a bit of trouble finding where the `.box-shadow()` mixin is defined in the less files. I generally use SCSS, but I did some quick research on LESS to see how mixins with parameters are declared. I don't see anything like this: ```.border-radius(@radius) {
  -webkit-border-radius: @radius;
     -moz-border-radius: @radius;
          border-radius: @radius;
}``` taken from http://lesscss.org/features/#mixins-parametric-feature
I figure I don't really _need_ to find that, but I want to understand what I am really doing before I start just ripping those out.
Could you help me find where that "function" is actually defined?

**Comment by thomaspark** [comment]
You should be able to find the definition at `bower_components/bootstrap/less/mixins/vendor-prefixes.less`.

Bootstrap 3.3.7 does need to be overwritten, but only uses the webkit prefix. We could drop the mixins by writing out `-webkit-box-shadow` and `box-shadow`, planning to drop webkit in the future.

**Comment by stphane** [comment]
I encountered this issue using v3.3.7 SCSS "Lumen" theme (along with a PHP scss parser, neither grunt nor Less). Unfortunately I could not get ride of this error without breaking the theme. Imports statements are as follows and seam to be ordered the right way
```
@import "../../web/bootswatch/lumen/_variables.scss";
@import "../../web/bootswatch/lumen/_bootstrap.scss";
@import "../../web/bootswatch/lumen/_bootswatch.scss";
```
Is there any place where I could find the updated scss files ? _Thank you_

**Comment by thomaspark** [comment]
Hey @stphane, it sounds like you're doing it correctly. The second line should refer to `_bootstrap.scss` from `bootstrap-sass-official`, as it doesn't exist under lumen. All should be well!

**Comment by stphane** [comment]
@thomaspark Thank you for your time. As you can see I've edited my answer few minutes after. I should now be set up !

**Comment by b2e9f523** [comment]
I just had the same issue when using Bootstrap and Bootswatch, installed via [Rails Assets](https://rails-assets.org/), with Rails 5.0

I did the following to resolve the issue and hope I helps somebody else:

1. **Add to Gemfile**
```ruby
source 'https://rails-assets.org' do
  gem 'rails-assets-bootstrap-sass-official', '3.3.7'
end
```
2. **Copy Bootswatch Theme files from Github into vendor/assets folder**
- `vendor/assets/stylesheets/bootswatch/[theme_name]/_variables.scss`
- `vendor/assets/stylesheets/bootswatch/[theme_name]/_bootswatch.scss`
3. **Use `@import` in Manifest file (application.css.scss)**
```ruby
@import "bootswatch/[theme_name]/variables";
@import "bootstrap-sass-official";
@import "bootswatch/[theme_name]/bootswatch";
```

**Comment by uxgnod** [comment]
same problem and solved with @b2e9f523 's way, but don't know why...

**Comment by Jtothec13** [comment]
@https://docs.google.com/document/d/1-gpNigK6jjW7UP-ixJeqH9vij6Ft5IMAoQHC5vRMAQw/edit
